packages/api/src/routes/medical/internal-cq.ts

```diff
@@ -209,6 +209,8 @@ router.post(
 
     const status = getPDResultStatus({ patientMatch: response.patientMatch });
 
+    response.duration = dayjs(response.responseTimestamp).diff(response.requestTimestamp);
+
```

Comment:
Doing this calculation here instead of Mirth, was taking long in rhino js and felt like it was better to do it here anyways

Reply:
Why don't we just get the current timestamp here, instead of introducing a new prop on IHE GW (responseTimestamp)?

Reply:
responseTimestamp already exists im introducing `requestTimestamp`

Reply:
That's what I mean - I'm talking about why introducing a new prop.

Reply:
In my mind it seemed more accurate to do it right before we make the request to the GW and right after to avoid any unexpected issues

Reply:
What unexpected issues? It seems pretty safe to:
- set the requested timestamp right before calling the IHE GW
- set the response timestamp right when we get the request back from IHE GW

It simplifies the logic by avoiding new props and less code on IHE GW.

What problems to you see with this approach - or why do you think its better to set the response timestamp on the IHE GW?

Reply:
Well its because for example Doc retrieval sends the document to s3 before sending it our api. So it wouldnt be a totally accurate representation of the time it takes to query the GW

Reply:
It depends on what we're trying to measure. I'd say that storing the result is part of the action (DR).

Also, I think this can get very specific (exactly what to include on the duration), but the main thing is to look at it over time, so it doesn't matter much if it includes sub-step X or Z, as long as we keep measuring the same thing.

I'd go with simplicity for this.

Reply:
Taken to Slack: https://metriport.slack.com/archives/C04CXLSAF7V/p1713981021786029?thread_ts=1713825972.462759&cid=C04CXLSAF7V

Moving forward with the current approach.